FEXCore/Source/Interface/Core/CodeCache.cpp

```diff
@@ -3,10 +3,135 @@
 
 #include <FEXCore/HLE/SourcecodeResolver.h>
 
+#include <FEXHeaderUtils/Filesystem.h>
+
+#include <Common/FDUtils.h>
+
+#include <fstream>
+
+#include <xxhash.h>
+
 namespace FEXCore {
 
 ExecutableFileInfo::~ExecutableFileInfo() = default;
 
+fextl::string CodeMap::GetBaseFilename(const ExecutableFileInfo& MainExecutable, bool AddNombSuffix) {
+  auto FileId = MainExecutable.FileId;
+
+  std::string_view base_filename = FHU::Filesystem::GetFilename(std::string_view {MainExecutable.Filename});
+  if (FileId != 0xffff'ffff'ffff'ffff) {
+    return fextl::fmt::format("{}-{:016x}{}", base_filename, MainExecutable.FileId, AddNombSuffix ? "-nomb" : "");
+  }
+
+  return "";
+}
+
+fextl::map<CodeMapFileId, CodeMap::ParsedContents> CodeMap::ParseCodeMap(std::ifstream& File) {
```

Comment:
This API usage will need to be changed before merging occurs. https://github.com/FEX-Emu/FEX/blob/main/docs/ProgrammingConcerns.md#stdfstream